VideoLocker/src/main/java/org/edx/mobile/module/notification/ParseNotificationDelegate.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class ParseNotificationDelegate implements NotificationDelegate{
+    private static final Logger logger = new Logger(ParseNotificationDelegate.class.getName());
+
+    ParseNotificationDelegate() {
+    }
+
+    /**
+     * if a user logs out, they should not receive the push notification
+     */
+    public void unsubscribeAll() {
+        new ParseSyncTask(MainApplication.instance().getApplicationContext()) {
+            @Override
+            public void onFinish(Void result) {
+                if (subscribedChannels != null) {
+                    for (String channel : subscribedChannels) {
+                        UserNotificationManager.instance.subscribeAndUnsubscribeToParse(channel, false);
+                    }
+                }
+            }
+        }.execute();
+    }
+
+    /**
+     * if user login,  he should subscribe channels based on local pref setting.
+     */
+    public void resubscribeAll() {
+        UserBasedPrefManager prefManager = UserBasedPrefManager.getInstance(UserBasedPrefManager.UserPrefType.NOTIFICATION);
+        NotificationPreference preference = prefManager.getNotificationPreference();
+        for (EdxLocalParseChannel pc : preference) {
+            if (pc.isSubscribed())
+                subscribeAndUnsubscribeToNotificationServer(pc.getChannelId(), true);
+        }
+    }
+
+    //check if local subscribed is not in parse server
+    //then try to subscribe it
+    public void syncWithNotificationServer() {
+        new ParseSyncTask(MainApplication.instance().getApplicationContext()) {
+            @Override
+            public void onFinish(Void result) {
+                UserBasedPrefManager prefManager = UserBasedPrefManager.getInstance(UserBasedPrefManager.UserPrefType.NOTIFICATION);
+                NotificationPreference preference = prefManager.getNotificationPreference();
+                for (EdxLocalParseChannel pc : preference) {
+                    if (pc.isSubscribed() && subscribedChannels != null &&
+                            !subscribedChannels.contains(pc.getChannelId()))
+                        subscribeAndUnsubscribeToNotificationServer(pc.getChannelId(), true);
+                    if (!pc.isSubscribed() && subscribedChannels != null &&
+                            subscribedChannels.contains(pc.getChannelId()))
+                        subscribeAndUnsubscribeToNotificationServer(pc.getChannelId(), false);
+                }
+            }
+        }.execute();
+
+    }
+
+    /**
+     * sync with current course enrollment.
+     * based on current course enrollment, we subscribe/unsubscribe to the Parse
+     * and update the local preference.
+     *
+     * @param responses
+     */
+    public void checkCourseEnrollment(List<EnrolledCoursesResponse> responses) {
+        List<CourseEntry> activeList = new ArrayList<>();
+        for (EnrolledCoursesResponse response : responses) {
+            if (response.isCourse() && response.isIs_active()) {
```

Comment:
@BenjiLee  that's original code. I dont want to change it for now as we will introduce Retrofit and rewrite this layer of code